flytekit/common/types/impl/schema.py

```diff
@@ -947,7 +947,7 @@ def cast_to(self, other_type):
                         additional_msg="Cannot cast because a required column '{}' was not found.".format(k),
                         received_value=self
                     )
-                if v != self.type.sdk_columns[k]:
```

Comment:
should you first not check that isinstance(v, SdkType)?

Reply:
sure can you re-approve?